src/asl/synth.ts

```diff
@@ -1771,6 +1773,15 @@ export class ASL {
             jsonPath: `${tempHeap}.string`,
           },
         };
+
+        // https://docs.aws.amazon.com/step-functions/latest/dg/amazon-states-language-intrinsic-functions.html#amazon-states-language-intrinsic-functions-escapes
+        function escapeFormatString(literal: ASLGraph.LiteralValue) {
+          if (typeof literal.value === "string") {
+            return literal.value.replace(/[\}\{\'}]/g, "\\$&");
```

Comment:
I don't understand what this fixes

Reply:
Escaping special characters (see link above)


`{${a}}` => 
   Before - `States.Format('{{}}', $.a)` - fails because `{` and `}` are special characters and `{{}}` is invalid.
   After - `States.Format('\{{}\}', $.a)` - succeeds!  `{helloworld}`

Reply:
How does that regex achieve that?

Reply:
\\$& ???

Reply:
Added a comment.